src/ChannelServer/World/Map.cs

```diff
@@ -199,13 +199,13 @@ public void RemoveMonster(Monster monster)
 		/// <param name="z"></param>
 		/// <param name="range"></param>
 		/// <returns></returns>
-		public List<Monster> GetAttackableMonstersInRange(float x, float y, float z, int range)
+		public IEnumerable<Monster> GetAttackableMonstersInRange(float x, float y, float z, int range)
 		{
 			var position = new Position(x, y, z);
 
-			List<Monster> result;
+			IEnumerable<Monster> result;
 			lock (_monsters)
-				result = _monsters.Values.Where(a => a.Position.InRange2D(position, range) && a.NpcType == NpcType.Monster).ToList();
+				result = _monsters.Values.Where(a => a.Position.InRange2D(position, range));
```

Comment:
Quick tip: You could simply return from inside the lock if there's nothing after it.